components/ecu/ecu_firmware/Core/Src/can.c

```diff
@@ -4,21 +4,31 @@
  * 
  * @date 2021/01/30
  * @author Blake Shular (blake-shular)
+
+ * @param currentVal the current flowing through the ECU
+ * @param doc_coc Discharge and Charge overcurrent status
  */
 
 #include "can.h"
+#include "main.h"
+
+#define CAN_ID 0x450
+#define MESSAGE_LENGTH 3
+
 
 CAN_TxHeaderTypeDef current_message_header = {
-    0x624, 0, CAN_ID_STD, CAN_RTR_DATA, 6, DISABLE
+    CAN_ID, 0, CAN_ID_STD, CAN_RTR_DATA, MESSAGE_LENGTH, DISABLE
 }; 
 
-void CAN_send_current(int currentVal) {
+void CAN_send_current(int currentVal, int doc_coc) {
     uint32_t begin_tick = HAL_GetTick();
     uint32_t * pTxMailbox = 0;
     HAL_StatusTypeDef status;
     currentVal /= 100;
```

Comment:
You still need to change this scaling according to our discussion, copied below for reference:

> For the new scaling:
>
> - The current value field in the CAN message data is 8 bits wide
> - The range of currents that must be representable by that field is ±75A
> - Your input to the function is 100ths of an amp (ie. if currentVal is -125 then the current flowing is -1.25A)
>
> The maximum value for a signed 8 bit value is 0b01111111 = 0x7F = 127. You can therefore calculate the scale factor you need as (1A/100)*(127/75A)